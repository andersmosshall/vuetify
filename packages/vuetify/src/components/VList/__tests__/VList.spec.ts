// @ts-nocheck
/* eslint-disable */

import {
  mount,
  Wrapper,
  MountOptions,
} from '@vue/test-utils'
// import VList from '../VList'
// import { ExtractVue } from '../../../util/mixins'

describe.skip('VList.vue', () => {
  type Instance = ExtractVue<typeof VList>
  let mountFunction: (options?: MountOptions<Instance>) => Wrapper<Instance>
  beforeEach(() => {
    mountFunction = (options?: MountOptions<Instance>) => {
      return mount(VList, {
        // https://github.com/vuejs/vue-test-utils/issues/1130
        sync: false,
        ...options,
      })
    }
  })

  it('should render component and match snapshot', () => {
    const wrapper = mountFunction()

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render loading list', () => {
    const wrapper = mountFunction({
      propsData: {
        loading: true,
      },
      mocks: {
        $vuetify: {
          rtl: false,
        },
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render list, which is link', () => {
    const wrapper = mountFunction({
      // https://github.com/vuejs/vue-test-utils/issues/1130
      sync: false,
      listeners: {
        click: () => {},
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render list with img', () => {
    const wrapper = mountFunction({
      propsData: {
        img: 'image.jpg',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render a flat list', () => {
    const wrapper = mountFunction({
      propsData: {
        flat: true,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render a raised list', () => {
    const wrapper = mountFunction({
      propsData: {
        raised: true,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render a list with custom height', async () => {
    const heightpx = '400px'
    const wrapper = mountFunction({
      // https://github.com/vuejs/vue-test-utils/issues/1130
      sync: false,
      propsData: {
        height: heightpx,
      },
    })

    expect(wrapper.element.style.height).toBe(heightpx)
    expect(wrapper.html()).toMatchSnapshot()

    wrapper.setProps({
      height: 401,
    })
    await wrapper.vm.$nextTick()
    expect(wrapper.element.style.height).toBe('401px')
  })
})
