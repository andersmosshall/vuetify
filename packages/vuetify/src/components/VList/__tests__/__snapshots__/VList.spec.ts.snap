// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VList.vue should render a list with custom height 1`] = `
<div class="v-list v-sheet theme--light"
     style="height: 400px;"
>
</div>
`;

exports[`VList.vue should render a flat list 1`] = `
<div class="v-list v-list--flat v-sheet theme--light">
</div>
`;

exports[`VList.vue should render a raised list 1`] = `
<div class="v-list v-list--raised v-sheet theme--light">
</div>
`;

exports[`VList.vue should render list with img 1`] = `
<div class="v-list v-sheet theme--light">
</div>
`;

exports[`VList.vue should render list, which is link 1`] = `
<div tabindex="0"
     class="v-list v-list--link v-sheet theme--light"
>
</div>
`;

exports[`VList.vue should render component and match snapshot 1`] = `
<div class="v-list v-sheet theme--light">
</div>
`;

exports[`VList.vue should render loading list 1`] = `
<div class="v-list v-list--loading v-sheet theme--light">
  <div class="v-list__progress">
    <div role="progressbar"
         aria-valuemin="0"
         aria-valuemax="100"
         class="v-progress-linear v-progress-linear--absolute v-progress-linear--visible theme--light"
         style="height: 4px;"
    >
      <div class="v-progress-linear__background primary"
           style="opacity: 0.3; left: 0%; width: 100%;"
      >
      </div>
      <div class="v-progress-linear__buffer">
      </div>
      <div class="v-progress-linear__indeterminate v-progress-linear__indeterminate--active">
        <div class="v-progress-linear__indeterminate long primary">
        </div>
        <div class="v-progress-linear__indeterminate short primary">
        </div>
      </div>
    </div>
  </div>
</div>
`;
